enhancements/authentication/oauth-resource-storage.md

```diff
@@ -0,0 +1,197 @@
+e.g. in order to show that information via oc or console, or to logout certain tokens.
+
+### Goals
+
+- Change the `oauthaccesstokens` and `oauthauthorizetokens` storage format and schema such that
+  the etcd keys alone (which are not encrypted when encryption at REST is enabled) are not sensitive.
+- Allow console and oc to log out all other sessions.
+
+### Non-Goals
+
+## Proposal
+
+We extend the OAuthAccessToken and OAuthAuthorizeToken types in the following way:
+
+```go
+// OAuthAccessToken describes an OAuth access token
+type OAuthAccessToken struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+    ...
+    
+    // SecretToken is the second half of the secret information a user has to pass as bearer token
+    // in order to authorize with the API server. It is concatenated with the name of the 
+    // OAuthAccessToken object.
+    // For old tokens, this field might be empty.
+    // +optional
+    SecretToken string `json:"secretToken,omitempty" protobuf:"bytes,11,opt,name=secretToken"`
+}
+
+// OAuthAuthorizeToken describes an OAuth authorization token
+type OAuthAuthorizeToken struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+    ...
+    
+    // SecretToken is the second half of the secret information a user has to pass as bearer token
+    // in order to authorize with the API server. It is concatenated with the name of the 
+    // OAuthAuthorizeToken object.
+    // For old tokens, this field might be empty.
+    // +optional
+    SecretToken string `json:"secretToken,omitempty" protobuf:"bytes,11,opt,name=secretToken"`
+}
+```
+
+### User Stories [optional]
+
+#### Story 1
+
+As an attacker who got access to an etcd backup I do not get access to oauth tokens.
+
+#### Story 2
+
+As a user I can use the API via oc or console to log out other sessions. 
+
+#### Story 3
+
+(optional) As a user I can list my other oauth sessions, including information like client name and expiration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The oauth authorizer in kube-apiserver in v4.6 will split the provided bearer token into 
+the first 43 bytes (256 bits, compare https://github.com/openshift/oauth-server/blob/3539e4065084dc0b5f35e43d7ea49631df99760d/pkg/server/crypto/random.go#L31-L35)
+and remaining bytes. It then looks up the `OAuthAccessToken` via the 
+former as a name (as we do today) and in addition compares the `SecretToken` field with
+the latter (possible zero length).   
+
+Old tokens will have an empty `SecretToken` field, and clients using that token will only
+pass 43 bytes bearer token, leading into the zero length case.
+
+Old kube-apiserver (on downgrade) will try to lookup the token object via the whole token
+and fail. I.e. on downgrade clients have to reauthenticate.
```

Comment:
This section should also include details on the tokens' creation and handling in the oauth-server as this would also need some changes, see the storage implementation we use to plug into osin:
https://github.com/openshift/oauth-server/blob/31279d2cabc12e26f7ab716b5f65e055214d372f/pkg/osinserver/registrystorage/storage.go

As an example that stands out, OAuthAuthorizeTokens' storage is specifically used during the authorization code grant in https://github.com/openshift/oauth-server/blob/31279d2cabc12e26f7ab716b5f65e055214d372f/vendor/github.com/RangelReale/osin/access.go#L183-L183.

These changes will most likely be small, yet worth mentioning.